### PR TITLE
Develop react_list-of-goods-hooks

### PR DESCRIPTION
**Pull request by sergey-lipikhin**
[DEMO LINK](https://sergey-lipikhin.github.io/react_list-of-goods-hooks/)

### CODE REVIEW

**Review by gretoq — APPROVED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -15,48 +16,115 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
```
Comment (sergii-nosachenko) [inline src/App.tsx:22]: Always add values for enum, it's safer for further development.

File: src/App.tsx [hunk 2]
```diff
@@ -15,48 +16,115 @@ export const goodsFromServer = [
...
+}
+
+function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) {
```
Comment (sergii-nosachenko) [inline src/App.tsx:29]: Add returning type too

File: src/App.tsx [hunk 3]
```diff
@@ -15,48 +16,115 @@ export const goodsFromServer = [
...
-          Reset
-        </button>
+        {isResetButtonVisible && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortType(SortType.NONE);
+              setIsReversed(false);
+            }}
```
Comment (sergii-nosachenko) [inline src/App.tsx:112]: Create a reset handler

File: src/App.tsx [hunk 4]
```diff
@@ -15,48 +16,115 @@ export const goodsFromServer = [
...
-        </ul>
+        {reorderedGoods.map(good => (
+          <li
+            data-cy="Good"
+            key={good}
+          >
+            {good}
+          </li>
+        ))}
       </ul>
```
Comment (sergii-nosachenko) [inline src/App.tsx:128]: Create a separate component

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]
Quick fixes are needed

**Review by sergii-nosachenko — APPROVED** [review]
Good work 👍

**Pull request closed without merging**